docs/tutorials/gluon/learning_rate_finder.md

```diff
@@ -0,0 +1,321 @@
+        
+    def plot(self):
+        lrs = [e[0] for e in self.results]
+        losses = [e[1] for e in self.results]
+        plt.figure(figsize=(6,8))
+        plt.scatter(lrs, losses)
+        plt.xlabel("Learning Rate")
+        plt.ylabel("Loss")
+        plt.xscale('log')
+        plt.yscale('log')
+        axes = plt.gca()
+        axes.set_xlim([lrs[0], lrs[-1]])
+        y_lower = min(losses) * 0.8
+        y_upper = losses[0] * 4
+        axes.set_ylim([y_lower, y_upper])
+        plt.show()
+```
+
+
+You can define the `LRFinderStoppingCriteria` as you wish, but empirical testing suggests using a smoothed average gives a more consistent stopping rule (see `smoothing`). We stop when the smoothed average of the loss exceeds twice the initial loss, assuming there have been a minimum number of iterations (see `min_iter`).
+
+
+```python
+class LRFinderStoppingCriteria():
+    def __init__(self, smoothing=0.3, min_iter=20):
+        """
+        smoothing: applied to running mean which is used for thresholding (float)
+        min_iter: minimum number of iterations before early stopping can occur (int)
+        """
+        self.smoothing = smoothing
+        self.min_iter = min_iter
+        self.first_loss = None
+        self.running_mean = None
+        self.counter = 0
+        
+    def __call__(self, loss):
+        """
+        loss: from single iteration (float)
+        """
+        self.counter += 1
+        if self.first_loss is None:
+            self.first_loss = loss
+        if self.running_mean is None:
+            self.running_mean = loss
+        else:
+            self.running_mean = ((1 - self.smoothing) * loss) + (self.smoothing * self.running_mean)
+        return (self.running_mean > self.first_loss * 2) and (self.counter >= self.min_iter)
+```
+
+## Usage
+
+Using a Pre-activation ResNet-18 from the Gluon model zoo, we instantiate our Learner and fire up our Learning Rate Finder!
+
+
+```python
+ctx = mx.gpu() if mx.test_utils.list_gpus() else mx.cpu()
+net = mx.gluon.model_zoo.vision.resnet18_v2(classes=10)
+learner = Learner(net=net, data_loader=data_loader, ctx=ctx)
+lr_finder = LRFinder(learner)
+lr_finder.find(lr_start=1e-6)
+```
+
+
+![png](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/doc/tutorials/lr_finder/finder_plot.png) <!--notebook-skip-line-->
+
+
+As discussed before, we should select a learning rate where the loss is falling (i.e. from 0.001 to 0.05) but before the loss starts to diverge (i.e. 0.1). We prefer higher learning rates where possible, so we select an initial learning rate of 0.05. Just as a test, we will run 500 epochs using this learning rate and evaluate the loss on the final batch. As we're working with a single batch of 128 samples, the variance of the loss estimates will be reasonably high, but it will give us a general idea. We save the initialized parameters for a later comparison with other learning rates.
+
+
+```python
+learner.net.save_params("net.params")
```

Comment:
`save_params` is deprecated. Should we use `save_parameters` instead?

Reply:
Changed.